frontend/public/components/storage/attach-storage.tsx

```diff
@@ -0,0 +1,437 @@
+/* eslint-disable no-undef */
+import * as _ from "lodash-es";
+import * as React from "react";
+import { Helmet } from "react-helmet";
+import { Link } from "react-router-dom";
+import {
+  k8sGet,
+  k8sUpdate,
+  K8sResourceKind,
+  referenceFor,
+} from "../../module/k8s";
+import * as k8sModels from "../../models";
+import {
+  ButtonBar,
+  history,
+  resourceObjPath,
+  ListDropdown,
+  ResourceIcon
+} from "../utils";
+import { formatNamespacedRouteForResource } from "../../ui/ui-actions";
+import {
+  getMountPaths,
+  createVolumeMount,
+  createVolume,
+  getKindFromPlural
+} from "../storage/storage";
+
+const Checkbox = ({ value, checked, onChange }) => (
+  <div>
+    <label className="checkbox-label">
+      <input
+        type="checkbox"
+        onChange={({ target: { checked: newChecked } }) =>
+          onChange(value, newChecked)
+        }
+        checked={!!checked}
+      />
+      &nbsp;&nbsp;
+      {value}
+    </label>
+  </div>
+);
+
+export const PvcDropdown = props => {
+  const kind = "PersistentVolumeClaim";
+  const namespace = props.namespace;
+  const resources = [{ kind, namespace }];
+  const selectedKey = props.selectedKey;
+  return (
+    <ListDropdown
+      {...props}
+      desc="Storage"
+      resources={resources}
+      selectedKeyKind={kind}
+      placeholder="Add or create storage"
```

Comment:
Let's leave this as "Add storage" if we don't support create yet. I would also make it, "Select persistent volume claim" We have been trying to call things what they are.